classes/table/files_table.php

```diff
@@ -0,0 +1,101 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Missing files table.
+ *
+ * @package   tool_objectfs
+ * @author    Dmitrii Metelkin <dmitriim@catalyst-au.net>
+ * @copyright Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_objectfs\table;
+
+defined('MOODLE_INTERNAL') || die();
+
+class files_table extends \table_sql {
+
+    /**
+     * Constructor.
+     *
+     * @param string $uniqueid
+     * @param string $objectlocation Status of the objects. Should be one of OBJECT_LOCATION_*
+     */
+    public function __construct($uniqueid, $objectlocation) {
+        parent::__construct($uniqueid);
+
+        $fields = 'f.*';
+        $from = '{files} f LEFT JOIN {tool_objectfs_objects} o on f.contenthash = o.contenthash';
+        $where = 'o.location = ?';
+        $params = [$objectlocation];
+
+        $this->columns = $this->headers = ['id', 'contextid', 'contenthash', 'path', 'component',
+            'filearea', 'filename', 'mimetype', 'filesize', 'timecreated'];
+
+        $this->define_columns($this->columns);
+        $this->define_headers($this->headers);
+
+        $this->set_sql($fields, $from, $where, $params);
+        $this->set_count_sql("SELECT COUNT(*) FROM $from WHERE $where", $params);
+
+        $this->downloadable = true;
+    }
+
+    public function col_id(\stdClass $row) {
+        return $row->id;
+    }
+
+    public function col_contextid(\stdClass $row) {
+        return $row->contextid;
+    }
+
+    public function col_contenthash(\stdClass $row) {
+        return $row->contenthash;
+    }
+
+    public function col_path(\stdClass $row) {
```

Comment:
Don't see the point of this column - also if included it should really be local file path as there's no guarantee that it will be stored in this format on all external file systems. 

mdl_files.filepath does have meaning though, i would show that here instead.

Reply:
That was a client requirement. I'll rename to "local file path "